src/app/sessions/services/sessions-creation.service.ts

```diff
@@ -31,9 +32,13 @@ export class SessionsCreationService {
         session.userTransactionId = transactionId;
         let transaction = this.createTransaction(session.id, transactionId);
 
-        this.store.dispatch(new SessionCreationActions.InitializeTransaction(transaction));
-        this.store.dispatch(new SessionActions.Update(session));
-        this.store.dispatch(new ProblemsActions.RemoveAll());
+        this.store.select(fromSessions.getSessionById(session.id))
```

Comment:
To avoid creating such one-off subscriber you could consider getting the session from state at the level of effects using 'withLatestFrom' operator. Please refer to: getSessionAfterCommit in session.effects file for an example.